### PR TITLE
Cqrs mediatr

### PR DESCRIPTION
**Pull request by Jens497**
This pull request consist of the Mediatr and CQRS pattern.
Furthermore the connection to the azure database etashblised.

2 notes:
1. The password for database should possibly be stored somewhere else like an .env file.
2. The way the db context works currently with DI etc. might be changed later.

### CODE REVIEW

**Comment by Goju-Ryu** [comment]
I think this PR needs a gitignore file and a lot of gennerated files removed if we want to avoid regular merge conflicts due to recompilation of the project.

I'd suggest this [VisualStudio gitignore file](https://github.com/github/gitignore/blob/main/VisualStudio.gitignore) to be used and the ignored files to be removed from the git history to avoid future problems.

File: RoleplayingSchemaBackend/Data/Users.cs
Comment (Goju-Ryu) [inline RoleplayingSchemaBackend/Data/Users.cs:1]: I'm unsure if this applies with our setup, but aren't all fields required by default unless set to nullable  with the question marked type?

File: RoleplayingSchemaBackend/User.cs
Comment (Goju-Ryu) [inline RoleplayingSchemaBackend/User.cs:1]: I can't figure out why there are two nearly identical User classes, what is each used for?

**Review by Goju-Ryu — CHANGES_REQUESTED** [review]

File: RoleplayingSchemaBackend/Data/Users.cs
Comment (Jens497) [reply RoleplayingSchemaBackend/Data/Users.cs:1]: I think so yes, this is mostly to make the code readable

File: RoleplayingSchemaBackend/User.cs
Comment (Jens497) [reply RoleplayingSchemaBackend/User.cs:1]: The first one was for dummy tests, it can be removed.

**Pull request closed without merging**